examples/feature_showcase/modal.rs

```diff
@@ -15,3 +15,36 @@ pub async fn modal(ctx: poise::ApplicationContext<'_, Data, Error>) -> Result<()
 
     Ok(())
 }
+
+/// Tests the Modal trait with component interactions.
+///
+/// Should be both prefix and slash to make sure it works without any slash command interaction
+/// present.
+#[poise::command(prefix_command, slash_command)]
+pub async fn component_modal(ctx: crate::Context<'_>) -> Result<(), Error> {
+    ctx.send(|m| {
+        m.content("Click the button below to open the modal")
+            .components(|c| {
+                c.create_action_row(|a| {
+                    a.create_button(|b| {
+                        b.custom_id("open_modal")
+                            .label("Open modal")
+                            .style(poise::serenity_prelude::ButtonStyle::Success)
+                    })
+                })
+            })
+    })
+    .await?;
+
+    while let Some(mci) =
+        poise::serenity_prelude::CollectComponentInteraction::new(ctx.serenity_context())
+            .timeout(std::time::Duration::from_secs(120))
+            .filter(move |mci| mci.data.custom_id == "open_modal")
+            .await
+    {
+        let data =
+            poise::execute_modal_on_component_interaction::<MyModal>(ctx, mci, None, None).await?;
+        println!("Got data: {:?}", data);
+    }
+    Ok(())
+}
```

Comment:
Yea, makes sense to me to have these live in the same file.  Again just didn't want to complicate things, was mostly trying to maintain the existing structures.